compiler/src/dotty/tools/dotc/core/TypeComparer.scala

```diff
@@ -404,8 +404,9 @@ class TypeComparer(initctx: Context) extends ConstraintHandling {
         if (cls2.isClass) {
           if (cls2.typeParams.isEmpty) {
             if (cls2 eq AnyKindClass) return true
-            if (tp1.isRef(defn.NothingClass)) return true
+            if (tp1.isRef(NothingClass)) return true
             if (tp1.isLambdaSub) return false
+            if (cls2 eq AnyClass) return true
               // Note: We would like to replace this by `if (tp1.hasHigherKind)`
```

Comment:
This comment is now attached to the wrong line (it refers to isLambdaSub above)

Reply:
Addressed.